R/test-threads.R

```diff
@@ -330,4 +298,63 @@ delete_models <- function(.mods, .tags = "test threads", .force = FALSE){
 }
 
 
+#' Parse model files and overwrite estimation iterations
+#'
+#' @param .mods a bbi model object or list of model objects. Generally created by `test_threads()`.
+#' @param .max_eval Max number of iterations for NONMEM to run.
+#' @param .detect Estimation options that will be set to `.max_eval` if found.
+#'
+#' @importFrom readr parse_number
+#'
+#' @keywords internal
+adjust_estimation_options <- function(.mods, .max_eval, .detect = c("MAXEVAL", "NITER", "NBURN", "MAX")){
+
+  assert_true(all(.detect %in% c("MAXEVAL", "NITER", "NBURN", "MAX")))
+
+  if(!is.null(.max_eval)){
+    assert_int(.max_eval, lower = 1)
+    search_str <- paste(.detect, collapse = "|")
+    map(.mods, function(.mod){
+      mod_path <- get_model_path(.mod)
+      mod_lines <- mod_path %>% readLines() %>% suppressSpecificWarning("incomplete final line found")
+      str_line_loc <- which(grepl(search_str, mod_lines))
+
+      if(is_empty(str_line_loc)){
+        delete_models(.mods = .mods, .force = TRUE) %>% suppressMessages()
+        stop("Neither MAXEVAL or NITER were found in the ctl file. Please ensure one is provided.")
+      }
+
+      str_values <- str_split(mod_lines[str_line_loc], " ")
+      for(i in seq_along(str_values)){
+
+        prevent <- c("MAXEVAL|NITER", "MAXEVAL|MAX", "MAX|NITER") # options that cant be set for the same estimation method
+        for(j in seq_along(prevent)){
+          prevent_loc <- grepl(prevent[j], str_values[[i]])
+          if(length(prevent_loc[prevent_loc]) > 1){
+            delete_models(.mods = .mods, .force = TRUE) %>% suppressMessages()
+            options_drop <- str_split(prevent[j], "\\|")[[1]]
+            stop(glue("Both {options_drop[1]} and {options_drop[2]} were set for the same estimation method. Please ensure only one is set"))
+          }
+        }
```

Comment:
This section might want to be pulled out for its own function, but I think this logic is definitely necessary. It may be a bit clunky, but it makes it very easy to add new conditions in the future that cant overlap for the same estimation method